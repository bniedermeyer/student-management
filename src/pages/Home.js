import React from 'react';
import { Card, Pane, Heading, Paragraph } from 'evergreen-ui';
import StyledLink from '../components/hoc/StyledLink';
import image from '../assets/undraw_exams_g4ow.svg';

const Home = () => (
  <Pane>
    <Heading size={900} margin={50} is="h1">
      Welcome to the Student Management Suite!
    </Heading>
    <img height={250} src={image} alt="animation of students studying" />
    <Paragraph margin={50}>Please choose from the options below</Paragraph>

    <Pane
      height={'100%'}
      display="flex"
      justifyContent="center"
      className="cards"
    >
      <Card
        elevation={1}
        hoverElevation={4}
        padding={50}
        display="flex"
        justifyContent="center"
        alignItems="center"
        flexDirection="column"
        margin={25}
        background="blueTint"
      >
        <StyledLink to="/students">
          <Heading>Student Search</Heading>
        </StyledLink>
      </Card>
      <Card
        elevation={1}
        hoverElevation={4}
        padding={50}
        display="flex"
        justifyContent="center"
        alignItems="center"
        flexDirection="column"
        margin={25}
        background="blueTint"
      >
        <StyledLink to="/add-student">
          <Heading>Add a Student</Heading>
        </StyledLink>
      </Card>
    </Pane>
  </Pane>
);

export default Home;
