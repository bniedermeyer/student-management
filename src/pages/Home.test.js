import React from 'react';
import Home from './Home';
import renderer from 'react-test-renderer';

it('should render the home page', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
