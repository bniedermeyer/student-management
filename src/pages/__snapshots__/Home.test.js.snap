// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the home page 1`] = `
<div
  className="css-nil ðŸ“¦box-szg_border-box"
>
  <h1
    className="ðŸ“¦mt_50px ðŸ“¦mb_50px ðŸ“¦fnt-sze_35px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_40px ðŸ“¦ltr-spc_-0-2px ðŸ“¦fnt-fam_wm9v6k ðŸ“¦color_234361 ðŸ“¦ml_50px ðŸ“¦mr_50px ðŸ“¦box-szg_border-box"
  >
    Welcome to the Student Management Suite!
  </h1>
  <img
    alt="animation of students studying"
    height={250}
    src="undraw_exams_g4ow.svg"
  />
  <p
    className="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_50px ðŸ“¦mb_50px ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_21px ðŸ“¦ltr-spc_-0-05px ðŸ“¦ml_50px ðŸ“¦mr_50px ðŸ“¦box-szg_border-box"
  >
    Please choose from the options below
  </p>
  <div
    className="cards css-nil ðŸ“¦h_100prcnt ðŸ“¦dspl_flex ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box"
  >
    <div
      className="css-1jo0gwu ðŸ“¦bs_17bru36 ðŸ“¦bg_1vicuwj ðŸ“¦bblr_5px ðŸ“¦bbrr_5px ðŸ“¦btlr_5px ðŸ“¦btrr_5px ðŸ“¦pb_50px ðŸ“¦pl_50px ðŸ“¦pr_50px ðŸ“¦pt_50px ðŸ“¦dspl_flex ðŸ“¦just-cnt_center ðŸ“¦algn-itms_center ðŸ“¦flx-drct_column ðŸ“¦mb_25px ðŸ“¦ml_25px ðŸ“¦mr_25px ðŸ“¦mt_25px ðŸ“¦box-szg_border-box"
    >
      <a
        href="/students"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h2
          className="ðŸ“¦mt_0px ðŸ“¦mb_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦fnt-fam_b77syt ðŸ“¦color_234361 ðŸ“¦box-szg_border-box"
        >
          Student Search
        </h2>
      </a>
    </div>
    <div
      className="css-1jo0gwu ðŸ“¦bs_17bru36 ðŸ“¦bg_1vicuwj ðŸ“¦bblr_5px ðŸ“¦bbrr_5px ðŸ“¦btlr_5px ðŸ“¦btrr_5px ðŸ“¦pb_50px ðŸ“¦pl_50px ðŸ“¦pr_50px ðŸ“¦pt_50px ðŸ“¦dspl_flex ðŸ“¦just-cnt_center ðŸ“¦algn-itms_center ðŸ“¦flx-drct_column ðŸ“¦mb_25px ðŸ“¦ml_25px ðŸ“¦mr_25px ðŸ“¦mt_25px ðŸ“¦box-szg_border-box"
    >
      <a
        href="/add-student"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h2
          className="ðŸ“¦mt_0px ðŸ“¦mb_0px ðŸ“¦fnt-sze_16px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦fnt-fam_b77syt ðŸ“¦color_234361 ðŸ“¦box-szg_border-box"
        >
          Add a Student
        </h2>
      </a>
    </div>
  </div>
</div>
`;
